lib/puppetserver/ca/puppet_config.rb

```diff
@@ -72,17 +72,33 @@ def resolve_settings(overrides = {})
         ssldir = overrides.fetch(:ssldir, '$confdir/ssl')
         settings[:ssldir] = substitutions['$ssldir'] = ssldir.sub('$confdir', confdir)
 
+        certdir = overrides.fetch(:certdir, '$ssldir/certs')
+        settings[:certdir] = substitutions['$certdir'] = certdir.sub(unresolved_setting, substitutions)
+
         cadir = overrides.fetch(:cadir, '$ssldir/ca')
         settings[:cadir] = substitutions['$cadir'] = cadir.sub(unresolved_setting, substitutions)
 
+        # Note the call to hostname below, this is horrific....
+        settings[:certname] = substitutions['$certname'] = overrides.fetch(:certname, `hostname`.chomp)
+        server = overrides.fetch(:server, '$certname')
+        settings[:server] = substitutions['$server'] = server.sub(unresolved_setting, substitutions)
+
+        settings[:ca_server] = overrides.fetch(:ca_server, '$server').sub(unresolved_setting, substitutions)
+
+        settings[:masterport] = substitutions['$masterport'] = overrides.fetch(:masterport, '8140')
+        settings[:ca_port] = overrides.fetch(:ca_port, '$masterport').sub(unresolved_setting, substitutions)
```

Comment:
I had to update this class for the hostcert/crl/localcacert so I added `ca_{server,port}` too, but it might be out of scope

Reply:
I think we don't need the ```.sub(unresolved_setting, substitutions)``` above for :ca_port since it will go through the .gsub below, right?